service/builder/exporters_builder.go

```diff
@@ -332,3 +357,18 @@ func createMetricsExporter(factoryBase component.ExporterFactoryBase,
 	// use ExporterFactoryOld.CreateMetricsExporter.
 	return factoryBase.(component.ExporterFactoryOld).CreateMetricsExporter(logger, cfg)
 }
+
+// createLogExporter creates a data exporter based on provided factory type.
+func createLogExporter(
+	factoryBase component.ExporterFactoryBase,
+	logger *zap.Logger,
+	cfg configmodels.Exporter,
+) (component.LogExporter, error) {
+	if factory, ok := factoryBase.(component.LogExporterFactory); ok {
+		creationParams := component.ExporterCreateParams{Logger: logger}
+		ctx := context.Background()
+		return factory.CreateLogExporter(ctx, creationParams, cfg)
+	}
+
+	return nil, fmt.Errorf("exporter %q does not support data type %q", factoryBase.Type(), configmodels.LogsDataType)
```

Comment:
Use configerror.ErrDataTypeIsNotSupported?

Reply:
ErrDataTypeIsNotSupported is not very user friendly. I want to have a detailed error message that explains what is wrong.